### PR TITLE
modify cdc owner & scheduler

### PR DESCRIPTION
**Pull request by zeminzhou**
### What problem does this PR solve?

modify cdc's owner & scheduler codes for tikv cdc

### What is changed and how does it work?

/cdc/cdc/owner files and /cdc/cdc/scheduler files

### Code changes

- Has exported function/method change
- Has exported variable/fields change
- Has interface methods change
- Has persistent data change
- No code

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Side effects

- Possible performance regression, WHY: **TBD**
- Increased code complexity, WHY: **TBD**
- Breaking backward compatibility, WHY: **TBD**
- No side effects

### Related changes

- Need to cherry-pick to the release branch
- Need to update the documentation
- Need to be included in the release note
- No related changes

### CODE REVIEW

**Comment by zeminzhou** [comment]
@pingyu PTAL~

File: cdc/cdc/owner/changefeed.go [hunk 1]
```diff
@@ -73,38 +71,43 @@ type changefeed struct {
...
-	newSink func() DDLSink,
+	/*
+		newDDLPuller func(ctx cdcContext.Context, startTs uint64) (DDLPuller, error),
+		newSink func() DDLSink,
+	*/
 ) *changefeed {
 	c := newChangefeed(id, gcManager)
-	c.newDDLPuller = newDDLPuller
-	c.newSink = newSink
+	/*
```
Comment (zz-jason) [inline cdc/cdc/owner/changefeed.go:107]: can you directly remove these codes?

**Review by zz-jason — COMMENTED** [review]
there are lots of code removals, is it possible to split this PR into two smaller PRs to save the reviewers and increase the PR merge speed:

1. one for cleaning codes, remove unused codes for TiKV CDC.
2. one for the functional changes to owner and scheduler

File: cdc/cdc/owner/changefeed.go (hunk 1 above)
Comment (zeminzhou) [reply cdc/cdc/owner/changefeed.go:107]: > can you directly remove these codes?

The way to comment the code is for the convenience of review, I will delete these comments later.

Comment (zeminzhou) [reply cdc/cdc/owner/changefeed.go:107]: ok

File: cdc/cdc/owner/metrics.go [hunk 2]
```diff
@@ -18,60 +18,60 @@ import "github.com/prometheus/client_golang/prometheus"
...
 		})
-	ownerMaintainTableNumGauge = prometheus.NewGaugeVec(
+	ownerMaintainKeySpanNumGauge = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: "ticdc",
+			Namespace: "tikv-cdc",
 			Subsystem: "owner",
-			Name:      "maintain_table_num",
-			Help:      "number of replicated tables maintained in owner",
+			Name:      "maintain_keyspan_num",
```
[the remark this answers is not in the document]
Comment (zeminzhou) [reply cdc/cdc/owner/metrics.go:58]: yes

**Pull request closed without merging**